### PR TITLE
fix: add option for image size

### PR DESCRIPTION
**Pull request by yuanLeeMidori**
Hi, I add the feature to check different image sizes with keyboard shortcuts. This PR is for solving the issue #91. 

While the user switch to the image tab, 
`z l` change the search result to only large size;
`z e` change the search result to only medium size;  (`z m` has been taken by filtering by past month)
`z i` change the search result to only icon size.

Take your time to review my code and please let me know if there is anything I can improve. Thank you!

### CODE REVIEW

File: src/search_engines.js [hunk 1]
```diff
@@ -444,6 +444,79 @@ class GoogleSearch {
     }
     return false;
   }
+
+  changeImageSize(size) {
+    const openTool = document.querySelector(
+        '[class="PNyWAd ZXJQ7c"][jsname="I4bIT"]');
```
Comment (infokiller) [inline src/search_engines.js:450]: The class names and other attributes seem brittle (it's highly likely that Google will change them).
Did you try to find something that is less likely to break?
Maybe there isn't a good alternative, but I wanted to verify.

File: src/search_engines.js [hunk 2]
```diff
@@ -444,6 +444,79 @@ class GoogleSearch {
...
+          break;
+        default:
+          break;
+      }
+    } else {
+      // While sized is selected, the dropdown element changes.
+      // Expend dropdown and select the size again.
+      switch (size) {
+        case 'l':
+          if (dropDownWithSize.innerHTML == 'Large') {
```
Comment (infokiller) [inline src/search_engines.js:486]: Will checking `'Large'` works in other languages?

File: src/search_engines.js [hunk 3]
```diff
@@ -444,6 +444,79 @@ class GoogleSearch {
...
+            break;
+          } else {
+            dropDownWithSize.click();
```
Comment (infokiller) [inline src/search_engines.js:489]: Here and below you are calling `dropDownWithSize.click()` but `dropDownWithSize` may be undefined?

File: src/search_engines.js [hunk 4]
```diff
@@ -444,6 +444,79 @@ class GoogleSearch {
     }
     return false;
   }
+
+  changeImageSize(size) {
```
Comment (infokiller) [inline src/search_engines.js:448]: I think it's better to use as input to the function something more descriptive than `'i', 'l', 'e'`. Javascript doesn't have enums but you can simulate them: https://stackoverflow.com/q/287903/1014208

File: src/search_engines.js (hunk 2 above)
Comment (infokiller) [inline src/search_engines.js:486]: Here and below you are calling dropDownWithSize.innerHTML but dropDownWithSize may be undefined?

File: src/search_engines.js [hunk 5]
```diff
@@ -444,6 +444,79 @@ class GoogleSearch {
...
+    }
+    const large = document.querySelector(
+        '[class="MfLWbb"][aria-label="Large"]');
+    const medium = document.querySelector(
+        '[class="MfLWbb"][aria-label="Medium"]');
+    const icon = document.querySelector(
+        '[class="MfLWbb"][aria-label="Icon"]');
+    const dropDownWithSize = document.querySelector(
+        '[class="xFo9P r9PaP Fmo8N"][jsname="wLFV5d"]');
+    if (large != null && medium != null && icon != null) {
```
Comment (infokiller) [inline src/search_engines.js:467]: It seems possible to simplify the two switch statements. Not sure about the best way to do it, but maybe something like:

```javascript
switch(size) {
  case LARGE:
    if (dropDownWithSize != null) {
      if (dropDownWithSize.innerHTML == 'Large') {
        return;
      }
      if (large == null) {
        dropDownWithSize.click();
        // large = ...
      }
    }
    if (large != null) {
      large.click();
    }
  // ...
}
```

Now basically some of the code in the case can be extracted out to a function, say something like:
```javascript
const setImageSize = (dropDownWithSize, getButton) => {
  let button = getButton();
  if (dropDownWithSize != null && button == null) {
    dropDownWithSize.click();
    button = getButton();
  }
  if (button != null) {
    button.click();
  }
};

// ...


switch(size) {
  case LARGE:
    if (dropDownWithSize != null && dropDownWithSize.innerHTML == 'Large') {
      return;
    }
    setImageSize(dropDownWithSize, () => { 
      // return document...;
    });
    break;
  // ...
}

```

**Review by infokiller — CHANGES_REQUESTED** [review]

**Comment by infokiller** [comment]
@yuanLeeMidori no pressure, but just making sure you know I'm waiting for changes I requested in earlier comments

**Comment by yuanLeeMidori** [comment]
@infokiller Hi, thank you for the reminder. Sorry for keeping you wait. If you want, you can close this PR, and I'll send a new one with that meets these requirements once I get time to work on it. Thank you!

**Comment by infokiller** [comment]
Don't worry, there's no pressure. I just wanted to make sure you're not waiting for me :)

File: src/search_engines.js (hunk 1 above)
Comment (yuanLeeMidori) [reply src/search_engines.js:472]: Hi, I tried to find an alternative but can't find one, sorry about that.

**Comment by yuanLeeMidori** [comment]
@infokiller Hi! Sorry for keeping you wait. I've made some changes based on your requests. Please let me know if there is anything I can improve. Thank you so much.

**Comment by infokiller** [comment]
@yuanLeeMidori thanks a lot for these changes and your contribution! from a quick look it looks good, but I'll want to test it a bit on my computer. I'm a bit busy in the next few days, so I'll probably get to it in a week or so.

**Comment by yuanLeeMidori** [comment]
@infokiller No worries! Take your time. Thank you for giving me the chance and help me to improve my code.

**Review by infokiller — APPROVED** [review]

**Comment by infokiller** [comment]
Thanks a lot @yuanLeeMidori it looks great! I just merged it

**Comment by yuanLeeMidori** [comment]
@infokiller Thank you for the waiting and reviewing, I'm very grateful. Thank you for giving me the chance :)

**Pull request merged**